go-controller/pkg/ovn/ovn.go

```diff
@@ -504,6 +506,17 @@ func (oc *Controller) WatchNodes() error {
 	_, err := oc.watchFactory.AddNodeHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			node := obj.(*kapi.Node)
+			noHostSubnet, err := oc.noHostSubnet(node)
+			if err != nil {
+				logrus.Errorf("error with ignored node label: %v", err)
+			}
+			if noHostSubnet {
+				oc.lsMutex.Lock()
+				defer oc.lsMutex.Unlock()
+				oc.logicalSwitchCache[node.Name] = nil
```

Comment:
@JacobTanenbaum we should probably have a comment here about setting the `nil` value to specifically indicate this node should not have a hostsubnet.

Reply:
fixed